commit 3374a5359e52f1671d8f831d65827d5020fe2595
Author: Jia Tan <jiat0218@gmail.com>
Date:   Thu May 11 23:49:23 2023 +0800

    liblzma: Adds lzma_nothrow to MicroLZMA API functions.
    
    None of the liblzma functions may throw an exception, so this
    attribute should be applied to all liblzma API functions.

diff --git a/src/liblzma/api/lzma/container.h b/src/liblzma/api/lzma/container.h
index 7e4ca3bf..5a891da9 100644
--- a/src/liblzma/api/lzma/container.h
+++ b/src/liblzma/api/lzma/container.h
@@ -615,7 +615,8 @@ extern LZMA_API(lzma_ret) lzma_stream_buffer_encode(
  *                output space (6 bytes) to create a valid MicroLZMA stream.
  */
 extern LZMA_API(lzma_ret) lzma_microlzma_encoder(
-		lzma_stream *strm, const lzma_options_lzma *options);
+		lzma_stream *strm, const lzma_options_lzma *options)
+		lzma_nothrow;
 
 
 /************
@@ -992,4 +993,4 @@ extern LZMA_API(lzma_ret) lzma_stream_buffer_decode(
 extern LZMA_API(lzma_ret) lzma_microlzma_decoder(
 		lzma_stream *strm, uint64_t comp_size,
 		uint64_t uncomp_size, lzma_bool uncomp_size_is_exact,
-		uint32_t dict_size);
+		uint32_t dict_size) lzma_nothrow;